### PR TITLE
CNV 1.4 installation guide updates

### PR DESCRIPTION
**Pull request by ousleyp**
New deployment procedure, in progress. Also changed the repository name to reflect 1.4, though I don't know for certain that it's the correct name. Removed mention of 1.3 from Installation Guide only.

### CODE REVIEW

**Comment by ousleyp** [comment]
@openshift/team-documentation PTAL ❄️

File: modules/cnv_install_1_4.adoc
Comment (simon3z) [inline modules/cnv_install_1_4.adoc]: @ousleyp please do not navigate to `/usr/share/ansible/kubevirt-ansible` and edit the original `vars/cnv.yml` file.
We should probably ask to copy it in the local path (E.g. `cp /usr/share/ansible/kubevirt-ansible/vars/cnv.yml cnv-vars.yml` or something similar). Then it means you have to change also the `vim` and `ansible-playbook` commands accordingly, something like: `ansible-playbook -e @cnv-vars.yml /usr/share/ansible/kubevirt-ansible/playbooks/kubevirt.yml`

File: modules/cnv_install_1_4.adoc
Comment (simon3z) [inline modules/cnv_install_1_4.adoc]: @ousleyp please do not document as default practice to use `--connection=local` (cc @nellyc). You can mention it as a possible parameter but I'd like the common practice to be remote (as openshift-ansible).

**Review by simon3z — CHANGES_REQUESTED** [review]

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: We clearly should not edit an rpm-shipped file, but why not `cd` to shorten our paths? A user should be encourage to add her changes to a homedir file, and add `-e @~/local-cnv-vars.yml` to `-e @vars/all.yml -e @vars/cnv.yml`

Also, iirc, we should include vars/all.yml in the commandline.

I would make sure that mentioning `~/local-cnv-args.yml` is not on the main procedure. We should be working fine out of the box. Only users with special needs (e.g. off-internet) need to have this file.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [inline modules/cnv_install_1_4.adoc]: Multus, Open vSwitch (OVS) and SR-IOV container network interface plug-ins

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: yes. facepalm.

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: New proposed steps, based on feedback from @simon3z and @dankenigsberg (let me know how it looks):

- `yum install kubevirt-ansible`
- `cp /usr/share/ansible/kubevirt-ansible/vars/cnv.yml local-cnv-vars.yml`
- `vim local-cnv-vars.yml`
- `cd /usr/share/ansible/kubevirt-ansible`
- `ansible-playbook -e @~/local-cnv-vars.yml -e @vars/all.yml -e @vars/cnv.yml playbooks/kubevirt.yml`
- verify install by navigating to the web UI

Feedback also welcome from @rthallisey (and anyone else).

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: Sure, I will remove that.

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: Great, I'll remove the APB tool from the prerequisites. 

@dankenigsberg - should I also remove "User with `cluster-admin` privileges"? @simon3z mentioned via email that I may be able to remove the entire section about creating a cluster admin user. This would leave the 3.11 cluster as the only prerequisite.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: I think that the last `-e` wins, so `@~/local-cnv-vars.yml` should come last. If it is really a *copy* of cnv.yml,  you do not need cnv.yml.

you could say: "in case you need to change anything in cnv.yml, copy it to ~/cnv.yml, modify it and use -e ~/cnv.yml instead of -e vars/cnv.yml"

the important bit is to keep the plain installation short and sweet. I hope that most users would not need to edit cnv.yml.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: I think that we *still* need a cluster admin in order to deploy, and we need to `oc login` so we have her credentials. I don't understand how we can drop it.

File: modules/cnv_install_1_4.adoc
Comment (simon3z) [reply modules/cnv_install_1_4.adoc]: > but why not `cd` to shorten our paths?

@dankenigsberg I go by the principle to not leave the session *dirty* at the end of the documented steps.

There may be other documented actions in the next chapters that do not assume you're in `/usr/share/ansible/kubevirt-ansible`.

File: modules/cnv_install_1_4.adoc
Comment (lukas-bednar) [reply modules/cnv_install_1_4.adoc]: Yes.

File: modules/cnv_install_1_4.adoc
Comment (lukas-bednar) [reply modules/cnv_install_1_4.adoc]: Yes.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: How about proper bash programming with

`pushd <dir>`
do your things
`popd`

File: modules/cnv_install_1_4.adoc
Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: We will still need to login as an admin in order to be able to create Cluster scoped objects.  So I think you can remove the apb sections, but replace it with what Dan mentioned: ```oc login ....``` or a doc line that says something like ```You need to login as admin```

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: Making that change now - thanks Dan.

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: No worries :)

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: Adding the login command as the first step of the install procedure.

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: @dankenigsberg @lukas-bednar Where should I include that?

File: modules/cnv_install_1_4.adoc
Comment (kalexand-rh) [inline modules/cnv_install_1_4.adoc]: I'd say " A running {product-title} 3.11 cluster"

File: modules/cnv_install_1_4.adoc
Comment (kalexand-rh) [inline modules/cnv_install_1_4.adoc]: What am I supposed to customize in this file? I'd say something like "If your cluster \<requires or uses\> \<things you can customize here\>, modify the `/usr/share/ansible/kubevirt-ansible/vars/cnv.yml` file."

File: modules/cnv_install_1_4.adoc
Comment (kalexand-rh) [inline modules/cnv_install_1_4.adoc]: s/- If/** If 
s/are using/use

File: modules/cnv_install_1_4.adoc
Comment (kalexand-rh) [inline modules/cnv_install_1_4.adoc]: s/- If/** If

File: modules/cnv_install_1_4.adoc
Comment (kalexand-rh) [inline modules/cnv_install_1_4.adoc]: s/using your OpenShift/by using your {product-title}

File: modules/cnv_requirements.adoc
Comment (kalexand-rh) [inline modules/cnv_requirements.adoc]: s/In order to /To

**Review by kalexand-rh — COMMENTED** [review]
Couple of picks and a question, but overall LGTM.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: what does `{product-title}` renders into, in this context?
a perquisite of CNV is OCP, not CNV itself.

File: modules/cnv_install_1_4.adoc
Comment (rthallisey) [inline modules/cnv_install_1_4.adoc]: At this point, we should expect that either the user is in `/usr/share/ansible/kubevirt-ansible` or in a local copy of this dir.

Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: For remote installs, we're going to use the inventory file that kubevirt-ansible ships with.  The user can edit it as needed.  So add to every `ansible-playbook` command a `-i inventory`.

**Review by rthallisey — CHANGES_REQUESTED** [review]

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: Duplicating code is usually a bad practice. Why would we want to promote it?

The only thing that we can possibly support are few edits to cnv.yml. Modifying roles or playbook should not be documented, nor keeping a possibility-stale copy of them.

Comment (simon3z) [reply modules/cnv_install_1_4.adoc]: > Duplicating code is usually a bad practice.

Agree 100%, bad practice no doubts.

> The only thing that we can possibly support are few edits to cnv.yml. Modifying roles or playbook should not be documented, nor keeping a possibility-stale copy of them.

+1 :+1:

File: modules/cnv_install_1_4.adoc
Comment (ohadlevy) [reply modules/cnv_install_1_4.adoc]: I wonder, is there an option to use environment variables instead of copying the files?

File: modules/cnv_install_1_4.adoc
Comment (ohadlevy) [reply modules/cnv_install_1_4.adoc]: I tend to agree here, if we end up copying something, that's usually a bad practice (e.g. copy of all of kubevirt-ansible).
can you:
* explain the motivation to copy the files? is it just how ansible searches for variables? 
* explain how often does a user end up changing the defaults? 

my motivation is to reduce the permutation of testing / verification to a minimum, if we can be opinionated instead of expose all of the playbook internals, the better. after all, this is our "api" to the user.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: I don't think you can use env variables (it does not show on https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable )

but I'm with you on not copying cnv.yml AT ALL. We should just say:

In order to use a custom repository instead ofregistry.access.redhat.com, add `-e registry_ur=registry.example.com -e cnv_repo_tag=local-repo-tag-for-cnv` to the command line.

To test the still-unsupported SR-IOV add `-e +deploy_sriov_plugin=true`

File: modules/cnv_install_1_4.adoc
Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: @dankenigsberg All the playbooks and roles from `kubevirt.yml`.  This will look for the `kubevirt` role in `./roles` and in `/etc/ansible/roles`.  If you copy only cnv.yml, the roles will be in neither.  Perhaps the best solution here is to install all our roles and playbooks in the default ansible location.

```yaml
....
# Deploy kubevirt
- name: Deploy kubevirt role
  hosts: localhost
  connection: local
  gather_facts: False
  # unset http_proxy. required for running in the CI
  environment:
    http_proxy: ""
  roles:
    - role: kubevirt

# Deploy kubevirt web UI
- import_playbook: kubevirt_web_ui.yml
when: platform == "openshift"
...
```

Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: 1. I don't want to copy `cnv.yml`. I'd like users to either specify their modification in the command line or write their own variable file.
2. I don't understand why using a `cnv.yml` from my home dir can possibly cause playbooks/kubevirt.yml to change the paths it uses. Are you sure you're solving a real problem? can you share an error log?

File: modules/cnv_install_1_4.adoc
Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: using `-e xxxx=yyy` has highest precedence;  it's `extra vars`.

Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: > using `-e xxxx=yyy` has highest precedence; it's `extra vars`.

correct. that's why I want users to use it to overwrite values in the chipped all.yml and cnv.yml.

File: modules/cnv_install_1_4.adoc
Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: forget it.  My config file was changing the default path. Non-issue.

File: modules/cnv_install_1_4.adoc
Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: To close the loop on this discussion, I think we all agree that we want to minimize the problem space by lowering the amount of "supported customizations".  Using extra vars will avoid copying any files and give the user the flexibility needed to make kubevirt-anisble work for their environment.  With that being said, can I get an ack from folks on the __minimal__ supported extra vars?
@simon3z @ohadlevy @dankenigsberg 

```
# Container Registry String: docker_prefix:docker_tag
docker_prefix
docker_tag
```

Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: This is the complete list. Product management may ask you change it in the future. Even *I* may ask you to include deploy_sriov to this list in a near future. But currently this is it. Only two extra args are supported, and only for offline deployment.

File: modules/cnv_install_1_4.adoc
Comment (rthallisey) [inline modules/cnv_install_1_4.adoc]: We'll also want to document deletion of CNV.

```yaml
ansible-playbook -e @/usr/share/ansible/kubevirt-ansible/vars/all.yml -e 
@~/local-cnv.yml /usr/share/ansible/kubevirt-ansible/playbooks/kubevirt.yml -e apb_action=deprovision
```

File: modules/cnv_install_1_4.adoc
Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: Adding one more...
```yaml
# Container Registry String: docker_prefix:docker_tag
docker_prefix
docker_tag

apb_action
```

Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: I don't know what `apb_action` does. This should not be a verbatim table, just a plain paragraph telling in order to do X run with `-e apb_action=bla`.

Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: `apb_action=provision` and `apb_acton=deprovision`.  `apb_acton=deprovision` will be needed for deletion.

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: Thanks, Ryan. A few questions about deprovisioning:

1. Does it matter where the command is run?
2. Are there any prerequisites other than having CNV installed in the first place?
3. What's a good way to test that deprovisioning was successful? Is there output that states the success/failure? If so, please send it so I can use it as an example.

Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: @lukas-bednar can probably provide the `oc` command line for checking if there is any kubevirt image running on the cluster.

Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: > 1. Does it matter where the command is run?
>
It should be run in the same places as deploy.  That is either in `/usr/share/ansible/kubevirt-ansible` or some place in the user's HOME directory.

> 2. Are there any prerequisites other than having CNV installed in the first place?
>
Having CNV installed is the only prerequisite

> 3. What's a good way to test that deprovisioning was successful? Is there output that states the success/failure? If so, please send it so I can use it as an example.
>
The playbook ran without errors is a success, since it will wait until everything is cleaned up before moving on

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: Adding "-i inventory" to both install and uninstall ansible-playbook commands.  

> At this point, we should expect that either the user is in `/usr/share/ansible/kubevirt-ansible` or in a local copy of this dir.

Is this still the case since we're having users specify any modifications on the command line?

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: @rthallisey @dankenigsberg 

1. Can you clarify if registry_url and cnv_repo_tag are supported args, or only docker_prefix, docker_tag and apb_action? 

2. Can you give an example of why someone would want to change docker_prefix and docker_tag?

Thanks!

Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: > Can you clarify if registry_url and cnv_repo_tag are supported args, or only docker_prefix, docker_tag and apb_action?
>
I would stick with just `docker_prefix`, `docker_tag` and `apb_action` because `registry_url` and `cnv_repo_tag` are both covered by `docker_prefix`;  it simplifies things.

> Can you give an example of why someone would want to change docker_prefix and docker_tag?
>
If a user has an on premise image registry where they hold our images, the install will need to pull images from their internal registry rather than Red Hat's registry.

Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: > > Can you clarify if registry_url and cnv_repo_tag are supported args, or only docker_prefix, docker_tag and apb_action?
> 
> I would stick with just `docker_prefix`, `docker_tag` and `apb_action` because `registry_url` and `cnv_repo_tag` are both covered by `docker_prefix`; it simplifies things.

Have you tested your suggestion, @rthallisey ? I see that our code uses `registry_url` directly in several places.

> 
> > Can you give an example of why someone would want to change docker_prefix and docker_tag?
> 
> If a user has an on premise image registry where they hold our images, the install will need to pull images from their internal registry rather than Red Hat's registry.

this example does not explain why a customer in his right mind would like to change `docker_tag` from its sane default `v1.4.0`.

May I repeat my request to start with my text, and only fix errors in it - if spotted? my text desribes what cnv.yml does to differentiate from kubevirt. it is also what QE does to test from brew. Let us write it down and get going.

**Comment by ousleyp** [comment]
https://github.com/openshift/openshift-docs/commit/5153af094b5adbc111ca808979eadc131870d917

https://cnv-new-deploy--ocpdocs.netlify.com/openshift-enterprise/latest/cnv_install/cnv_install.html

@dankenigsberg @simon3z @rthallisey @nellyc 

Please review the latest commit/preview when you're able - thank you all for your help! 

I need further clarification on what needs to be done for remote install, but I incorporated the changes mentioned on this PR as I understood them. Please correct me if I misunderstood anything. :)

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: Done, @dankenigsberg - check out the latest commit and let me know what I got wrong. :)

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [inline modules/cnv_install_1_4.adoc]: I don't see the point of this note. kubevirt-ansible.rpm already requires these packages.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [inline modules/cnv_install_1_4.adoc]: I'd prefer to have `inventory_file`. but most importantly, I miss the content of this file. I don't recall if @lukas-bednar provided you an example.

**Review by dankenigsberg — CHANGES_REQUESTED** [review]

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [inline modules/cnv_install_1_4.adoc]: please start with this, as - if necessary - it would pull in the `oc` client.

**Review by dankenigsberg — CHANGES_REQUESTED** [review]

File: modules/cnv_install_1_4.adoc
Comment (nellyc) [reply modules/cnv_install_1_4.adoc]: we are not supposed to use all.yml, only cnv.yml

also, i believe that we should first cd into /usr/share/ansible/kubevirt-ansible
and execute the command from there in order to not need to be a privileged user

File: modules/cnv_uninstall_1_4.adoc
Comment (nellyc) [inline modules/cnv_uninstall_1_4.adoc]: above comment is relevant here as well

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: I'm pretty sure we *do* need to specify all.yml. cnv.yml overrides only *some* of the vars there.

Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: @nellyc @lukas-bednar don't you use all.yml in QE?

File: modules/cnv_install_1_4.adoc
Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: The web_ui doesn't use docker_prefix.  I must've missed it in my review.  Also, @dankenigsberg looks like network hardcodes the registry to upstream images.  I think these are both bugs @nellyc.
```yaml
multus_provisioner_repo: "docker.io/nfvpe/multus"
cni_provisioner_repo: "quay.io/schseba/cni-plugins"
ovs_provisioner_repo: "quay.io/kubevirt/ovs-cni-plugin"
```

Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: ignore my comment.  cnv.yaml covers it.

File: modules/cnv_install_1_4.adoc
Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: Update for the deploy command.
  - Remove all.yml extra_var
  - add apb_action=provision as an extra_var
```diff
-ansible-playbook -i <inventory file> -e @/usr/share/ansible/kubevirt-ansible/vars/all.yml -e @/usr/share/ansible/kubevirt-ansible/vars/cnv.yml  /usr/share/ansible/kubevirt-ansible/playbooks/kubevirt.yml
+ansible-playbook -i <inventory file> -e @/usr/share/ansible/kubevirt-ansible/vars/cnv.yml  /usr/share/ansible/kubevirt-ansible/playbooks/kubevirt.yml -e apb_action=provision
```

Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: @ousleyp  please wait for QE approval before dropping all.yml.

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: That makes sense. I'll remove it and switch the order of the "oc login" and package installation commands.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: Dropping `all.yml` is wrong, regardless to that BZ. Why? upstream deployment of KubeVirt includes `all.yml`. `cnv.yml` has downstream-only modifications for it. I expect that most often than not, new variables can be set in all.yml without cnv.yml caring about them. If you drop `all.yml`, cnv.yml would have to become a not-so-precise copy of it. That's a very bad practice - it would quietly break when upstream is modified.

We *may* be able to include `all.yml` from within `cnv.yml` to help shorten the command line. I do not know ansible well enough to answer that.

Adding `apb_action=provision` is functionally harmless, but I don't see the benefit thereof.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [inline modules/cnv_install_1_4.adoc]: @simon3z explicitly asked not to leave the reader inside this directory. We can either use full paths as he requested, or use `pushd /usr/share/ansible/kubevirt-ansible` here, and `popd` in the end as I suggested.

Can we please get this doc converged? It feels sad to see @simon3z's comment done and then undone.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [inline modules/cnv_install_1_4.adoc]: I asked *not* to drop all.yml before a strict ack from QE that this actually works. no such ack was received, so why is this dropped?

even if an ack is received, I think it is important to keep it.

File: modules/cnv_uninstall_1_4.adoc
Comment (dankenigsberg) [inline modules/cnv_uninstall_1_4.adoc]: why do you bother to `cd` above if you use the full path here?

**Review by dankenigsberg — CHANGES_REQUESTED** [review]
the content of `inventory_file` is still not explained.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: I just spoke with @nellyc , she told me that as of yesterday @lukas-bednar started running CI without `all.yml`. I still think it's wrong - we should do things upstream first - but you can remove `all.yml`.

File: modules/cnv_install_1_4.adoc
Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: > the content of `inventory_file` is still not explained.

... or at least I cannot see it. You may say:

See "references" for an example `inventory_file` .

Comment (rthallisey) [reply modules/cnv_install_1_4.adoc]: I think the problem is that cnv.yml morphed into a patch of all.yml.  Things like `kubevirt_web_ui_branding` and `deploy_sriov_plugin` were duplicated in cnv.yml.  Unfortunately, downstream doesn't have a code review process, so now cnv.yml covers the functionality of all.yml.

File: modules/cnv_uninstall_1_4.adoc
Comment (ousleyp) [reply modules/cnv_uninstall_1_4.adoc]: Nelly had said "also, i believe that we should first cd into /usr/share/ansible/kubevirt-ansible
and execute the command from there in order to not need to be a privileged user" in another comment. That's the only reason I added the `cd` command. But I can shorten those paths and use pushd/popd as you suggested. Will update the PR momentarily.

Comment (dankenigsberg) [reply modules/cnv_uninstall_1_4.adoc]: ok. just make sure to do this with `pushd` and finish with `popd` to make @simon3z happy.

Comment (ousleyp) [reply modules/cnv_uninstall_1_4.adoc]: Please review the changes, @dankenigsberg: https://github.com/openshift/openshift-docs/pull/13328/commits/6010dcf48c2bb4c3d411f0062cb7045bf28be1d2

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: As mentioned in another comment, here are the latest updates - the user is no longer left in `kubevirt-ansible`: https://github.com/openshift/openshift-docs/pull/13328/commits/6010dcf48c2bb4c3d411f0062cb7045bf28be1d2

Comment (dankenigsberg) [reply modules/cnv_install_1_4.adoc]: thanks! I still find it hard to review stuff on github, I'm used to a more linear review.

**Review by dankenigsberg — APPROVED** [review]

File: modules/cnv_install_1_4.adoc
Comment (ousleyp) [reply modules/cnv_install_1_4.adoc]: No worries @dankenigsberg, it's extremely hard to keep track of what's being said, especially when force-pushing removes some comments. Thanks for the review!

File: modules/cnv_uninstall_1_4.adoc
Comment (nellyc) [inline modules/cnv_uninstall_1_4.adoc]: i would replace the 'by adding',
we have apb_action=provision when installing, 
so it should be 'by replacing the apb_action param to...'

Comment (ousleyp) [reply modules/cnv_uninstall_1_4.adoc]: Oh, good catch! I'll update that shortly and then move the BZ to ON_QA.

Comment (ousleyp) [reply modules/cnv_uninstall_1_4.adoc]: Nelly, how's this? https://github.com/openshift/openshift-docs/pull/13328/commits/1d86405db513624f8b4184dc6d743e85d59ea008

Comment (dankenigsberg) [reply modules/cnv_uninstall_1_4.adoc]: /lgtm for what it's worth.

**Review by dankenigsberg — APPROVED** [review]

**Review by rthallisey — APPROVED** [review]

**Pull request merged**
